src/main/kotlin/org/rust/ide/annotator/RsExpressionAnnotator.kt

```diff
@@ -81,8 +82,12 @@ private class RedundantParenthesisVisitor(private val holder: AnnotationHolder)
 
     private fun RsExpr?.warnIfParens(message: String) {
         if (this is RsParenExpr) {
-            holder.createWeakWarningAnnotation(this, message)
-                .registerFix(RemoveParenthesesFromExprIntention())
+            when (this.children.singleOrNull()) {
```

Comment:
I think this overshots a bit. Sometimes is totally legit to remove parenthesis around struct literals:

```
let s = (S { foo: 92 });
```

We need to check that the parent of the exression is a conditon/match target.